### PR TITLE
add a quick-start section to our docs

### PR DESCRIPTION
**Pull request by astoycos**
Allow users to get started quickly with bpfman
by either deploying from an RPM or in kubernetes.

Additionally changes the getting-started section
to a documentation section, and the developer
section to developer documentation section.

Fixup some other small things along the way.

### CODE REVIEW

File: docs/getting-started/running-rpm.md [hunk 1]
```diff
@@ -21,9 +21,26 @@ to be installed:
...
+To install officially released versions:
+
+```console
+sudo dnf copr enable @ebpf-sig/bpfman
+```
+
+To install nightly builds:
+
+```console
+sudo dnf copr enable @ebpf-sig/bpfman
```
Comment (anfredette) [inline docs/getting-started/running-rpm.md:35]: Is this supposed to be @ebpf-sig/bpfman-next?

File: docs/quick-start.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+2024/05/02 14:58:13 Program registered with id 18569
+2024/05/02 14:58:14 SIGUSR1 signal count: 8
+2024/05/02 14:58:15 SIGUSR1 signal count: 9
+2024/05/02 14:58:16 SIGUSR1 signal count: 10
+```
+
+## Deploy Released container images on Kubernetes
+
+The quickest solution for running `bpfman` in a Kubernetes deployment is to run a
+local Kubernetes KIND Cluster:
```
Comment (anfredette) [inline docs/quick-start.md:82]: I suggest you link KIND Cluster to https://kind.sigs.k8s.io/docs/user/quick-start/

**Review by anfredette — CHANGES_REQUESTED** [review]
I didn't run all the commands, but it looks good otherwise.

File: docs/getting-started/running-rpm.md (hunk 1 above)
Comment (astoycos) [reply docs/getting-started/running-rpm.md:35]: Yep good catch

File: docs/getting-started/running-rpm.md [hunk 3]
```diff
@@ -21,9 +21,26 @@ to be installed:
...
+
+> **Note:** If both the bpfman and bpfman-next copr repos are enabled DNF will
+> automatically pull from bpfman-next.  To disable one or the other simply run
+> ```console
+> sudo dnf copr disable @ebpf-sig/bpfman-next
+> ```
+
 ### Install RPM From Packit Service
 
 To load an RPM from a specific commit, find the commit from
```
[the remark this answers is not in the document]
Comment (astoycos) [reply docs/getting-started/running-rpm.md:46]: Done! I like this 👍

File: docs/quick-start.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
+# Bpfman Quick Start
```
[the remark this answers is not in the document]
Comment (astoycos) [reply docs/quick-start.md:1]: Done!

File: docs/quick-start.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+`bpfman-rpc`, `bpfman` Library and `bpfman-agent`:
+
+```console
+kubectl apply -f https://github.com/bpfman/bpfman/releases/download/v${BPFMAN_REL}/bpfman-operator-install-v${BPFMAN_REL}.yaml
+```
+
+Finally, deploy an example eBPF program.
+
+```console
+kubectl apply -f https://github.com/bpfman/bpfman/releases/download/v${BPFMAN_REL}/go-xdp-counter-install-v${BPFMAN_REL}.yaml
```
[the remark this answers is not in the document]
Comment (astoycos) [reply docs/quick-start.md:105]: Agreed for RC release, Please see the release PR I added a note to the RELEASE.md talking about RC releases and how they might not have all the correct artifacts.

Minor and Major releases should still have them

File: docs/quick-start.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+```
+
+There are other example programs in each [Release](https://github.com/bpfman/bpfman/releases)
```
[the remark this answers is not in the document]
Comment (astoycos) [reply docs/quick-start.md:108]: See above comment/ read the v0.4.1 release pr and let me know if that works :)

File: docs/quick-start.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/quick-start.md:105]: Sounds good.

File: docs/quick-start.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/quick-start.md:108]:
```suggestion
There are other example program install yamls in the artifacts for each [Release](https://github.com/bpfman/bpfman/releases)
```
This was going to be my initial comment until I couldn't find the yamls.

[the remark this answers is not in the document]
Comment (astoycos) [reply docs/quick-start.md]: Done!

**Review by anfredette — APPROVED** [review]
LGTM

**Review by Billy99 — APPROVED** [review]
LGTM

Thanks Andrew!

**Pull request merged**